targets/TARGET_TOSHIBA/TARGET_TMPM066/i2c_api.c

```diff
@@ -0,0 +1,361 @@
+typedef struct {
+    uint32_t sck;
+    uint32_t prsck;
+} I2C_clock_setting_t;
+
+static const uint32_t I2C_SCK_DIVIDER_TBL[8] = { 20, 24, 32, 48, 80, 144, 272, 528};  // SCK Divider value table
+static uint32_t start_flag = 0;
+I2C_clock_setting_t clk;
+I2C_State status;
+
+static int32_t wait_status(i2c_t *obj)
+{
+    volatile int32_t timeout = I2C_TIMEOUT;
+
+    while (I2C_GetINTI2CStatus(obj->i2c) == DISABLE) {
+        if ((timeout--) == 0) {
+            return (-1);
+        }
+    }
+    return (0);
+}
+
+// Initialize the I2C peripheral. It sets the default parameters for I2C
+void i2c_init(i2c_t *obj, PinName sda, PinName scl)
+{
+    MBED_ASSERT(obj != NULL);
+    I2CName i2c_sda = (I2CName)pinmap_peripheral(sda, PinMap_I2C_SDA);
+    I2CName i2c_scl = (I2CName)pinmap_peripheral(scl, PinMap_I2C_SCL);
+    I2CName i2c_name = (I2CName)pinmap_merge(i2c_sda, i2c_scl);
+    MBED_ASSERT((int)i2c_name != NC);
+
+    switch(i2c_name) {
+        case I2C_0:
+            CG_SetFcPeriphA(CG_FC_PERIPH_I2C0, ENABLE);
+            obj->i2c = TSB_I2C0;
+            obj->IRQn = INTI2C0_IRQn;
+            break;
+        case I2C_1:
+            CG_SetFcPeriphB(CG_FC_PERIPH_I2C1, ENABLE);
+            obj->i2c = TSB_I2C1;
+            obj->IRQn = INTI2C1_IRQn;
+            break;
+        default:
+            error("I2C is not available");
+            break;
+    }
+
+    pinmap_pinout(sda, PinMap_I2C_SDA);
+    pin_mode(sda, OpenDrain);
+    pin_mode(sda, PullUp);
+
+    pinmap_pinout(scl, PinMap_I2C_SCL);
+    pin_mode(scl, OpenDrain);
+    pin_mode(scl, PullUp);
+
+    NVIC_DisableIRQ(obj->IRQn);
+
+    i2c_reset(obj);
+    i2c_frequency(obj, 100000);
+}
+
+// Configure the I2C frequency
+void i2c_frequency(i2c_t *obj, int hz)
+{
+    uint64_t sck, tmp_sck;
+    uint64_t prsck, tmp_prsck;
+    uint64_t fscl, tmp_fscl;
+    uint64_t fx;
+
+    SystemCoreClockUpdate();
+
+    if (hz <= 1000000) {
+        sck   = tmp_sck   = 0;
+        prsck = tmp_prsck = 1;
+        fscl  = tmp_fscl  = 0;
+        for (prsck = 1; prsck <= 32; prsck++) {
+            fx = ((uint64_t)SystemCoreClock / prsck);
+            if ((fx < 40000000U) && (fx > 6666666U)) {
+                for (sck = 0; sck <= 7; sck++) {
+                    fscl = (fx / (uint64_t)I2C_SCK_DIVIDER_TBL[sck]);
+                    if ((fscl <= (uint64_t)hz) && (fscl > tmp_fscl)) {
+                        tmp_fscl = fscl;
+                        tmp_sck = sck;
+                        tmp_prsck = (prsck < 32)? prsck: 0;
+                    }
+                }
+            }
+        }
+        clk.sck = (uint32_t)tmp_sck;
+        clk.prsck = (tmp_prsck < 32) ? (uint32_t)(tmp_prsck) : 0;
+    }
+    obj->myi2c.I2CSelfAddr = SELF_ADDR;
+    obj->myi2c.I2CDataLen = I2C_DATA_LEN_8;
+    obj->myi2c.I2CACKState = ENABLE;
+    obj->myi2c.I2CClkDiv = clk.sck;
+    obj->myi2c.PrescalerClkDiv = clk.prsck;
+
+    I2C_Init(obj->i2c, &obj->myi2c);
+    NVIC_DisableIRQ(obj->IRQn);
```

Comment:
why is this disabling IRQ (related to slave functionality that operates with IRQs) ?